Makefrag

```diff
@@ -0,0 +1,43 @@
+# FLEXPRET_ROOT_DIR and NAME must be defined
+
+LIB_DIR := $(FLEXPRET_ROOT_DIR)/programs/lib
+LINKER_SCRIPT := $(LIB_DIR)/linker/flexpret.ld
+CC := riscv32-unknown-elf-gcc
+OBJDUMP := riscv32-unknown-elf-objdump
+OBJCOPY := riscv32-unknown-elf-objcopy
+EMU := $(FLEXPRET_ROOT_DIR)/emulator/fp-emu
+
+NUM_THREADS ?= 1 # default number of threads
+
+.PHONY: compile riscv dump mem run clean
+
+compile: riscv dump mem
+
+riscv: $(NAME).riscv
+%.riscv: *.c
```

Comment:
I think it would be better to define the following variables:
APP_SOURCES
LIB_SOURCES

The first one should be set in the Makefile in the folder of the application. The wildcard of using all .C files will only work for very simple examples. I would also put all the FP library into a variable instead of naming them expliciticly here. It is also quite standard to define a variable CFLAGS wher `-g -static -O0` etc goes.